### ISSUE TITLE
why does the loss never change during finetune

### ISSUE THREAD

**Issue by Annie1900** [magic-research/PLLaVA#6]
`2024-04-29T21:48:58 | utils.basic_utils: Train Epoch: [0]  [  0/336]  eta: 0:29:38  video-loss: 10.3755  time: 5.2940  data: 3.7708  max mem: 40052 res mem: 43552
2024-04-29T21:49:05 | utils.basic_utils: Train Epoch: [0]  [  5/336]  eta: 0:11:11  video-loss: 10.3755  time: 2.0277  data: 0.7843  max mem: 42609 res mem: 45852
2024-04-29T21:49:10 | utils.basic_utils: Train Epoch: [0]  [ 10/336]  eta: 0:08:34  video-loss: 10.3755  time: 1.5771  data: 0.4341  max mem: 42609 res mem: 45852`

**Comment by Annie1900** [comment]
Excuse me, I want to use a specific data set for fine-tuning, but loss remains consistent, the model doesn't seem to be training, I want to know why, here are my Settings
`from tasks.train.instruction_data import *

# ========================= data ==========================
# train_corpus = "videochat2_instruction"
train_corpus = "data_finance"

train_file = "${available_corpus[${train_corpus}]}"  # for lazy evaluation
test_file = dict()
test_types = []
num_workers = 8
save_steps=1000
ckpt_steps=1000
stop_key = None
deepspeed=False
# ========================= input ==========================
num_frames = 16
num_frames_test = 1
batch_size = 1
gradient_accumulation_steps=16
max_txt_l = 512
max_train_steps=None
pre_text = False
gradient_checkpointing=False
inputs = dict(
    image_res=336,
    video_input=dict(
        num_frames="${num_frames}",
        sample_type="rand",
        num_frames_test="${num_frames_test}",
        sample_type_test="middle",
        random_aug=False,
    ),
    max_txt_l=dict(image="${max_txt_l}", video="${max_txt_l}"),
    batch_size=dict(image="${batch_size}", video="${batch_size}"),
    batch_size_test=dict(image="${batch_size}", video="${batch_size}"),
)

# ========================= model ==========================
model = dict(
    repo_id="/u01/isi/luweiheng/pllava/PLLaVA/checkpoint/ermu2001/pllava-7b",
    pretrained_path="/u01/isi/luweiheng/pllava/PLLaVA/checkpoint/ermu2001/pllava-7b",
    load_from_origin=False,
    origin_vision="",
    origin_llm="",
    vision_encoder=dict(
        name="vit_l14", # somehow need this to tell the dataset the mean std of pretrained model
    ),
    torch_dtype='bfloat16',
    freeze_projector=False,
    freeze_lm=False,
    freeze_vision_tower=True,
    lora_target_modules=["q_proj", "v_proj"], # for llama/mistral/gemma
    use_lora=True,
    lora_r=128,
    lora_alpha=32,
    lora_dropout=0.05,
    num_frames="${num_frames}",
    pooling_method='avg',
    use_pooling=True,
    frame_shape=(24,24),
    pooling_shape=(16,8,8),
)
preprocess = dict(
    system="",
    mm_alone=True,
    random_shuffle=True,
    add_second_msg=True,
    roles=['USER:', 'ASSISTANT:'],
    end_signal=(' ', '</s>'),
    begin_signal='',
    dataset_image_placeholder='<Image></Image>',
    dataset_video_placeholder='<Video></Video>',
    image_token_index=32000,
    max_txt_l = "${max_txt_l}",
    ignore_index=-100, # same as torch softmax ignore index 
    center_pad=False,
    longest_edge=762,
    shortest_edge=336,
    clip_transform=False,
    num_frames="${num_frames}",
)


optimizer = dict(
    opt="adamW",
    lr=2e-5,
    opt_betas=[0.9, 0.999],  # default
    weight_decay=0.02,
    max_grad_norm=-1,  # requires a positive float, use -1 to disable
    # use a different lr for some modules, e.g., larger lr for new modules
    different_lr=dict(enable=False, module_names=[], lr=1e-3),
)

# scheduler = dict(sched="cosine", epochs=3, min_lr_multi=0.25, warmup_epochs=0.6)
# scheduler = dict(sched="cosine", epochs=3, min_lr_multi=0.25, warmup_epochs=0.6)
scheduler = dict(
    is_videochat2_custom=False,
    sched="cosine", 
    epochs=2, 
    warmup_ratio=0.2,
    min_lr_multi=0.25)

evaluate = False
deep_fusion = False
evaluation = dict(
    eval_frame_ensemble="concat",  # [concat, max, mean, lse]
    eval_x_only=False,
    k_test=128,
    eval_offload=True,  # offload gpu tensors to cpu to save memory.
)

fp16 = True
gradient_checkpointing = True

# ========================= wandb ==========================
wandb = dict(
    enable=False,
    entity="user",  # username or team name to store the runs, see https://docs.wandb.ai/ref/python/init
    project="videochat2",  # setup in your command line
)
dist_url = "env://"
device = "cuda"
mode = "it"

# ========================= others ==========================
output_dir = None  # output dir
resume = False  # if True, load optimizer and scheduler states as well
debug = False
log_freq = 5
metric_window_size=10 # window size for metric
seed = 42
report_to='tensorboard'
save_latest = True
auto_resume = True
pretrained_path = ""  # path to pretrained model weights, for resume only?`
`echo "PYTHONPATH: ${PYTHONPATH}"
which_python=$(which python)
echo "which python: ${which_python}"
export PYTHONPATH=${PYTHONPATH}:${which_python}
export PYTHONPATH=${PYTHONPATH}:.
echo "PYTHONPATH: ${PYTHONPATH}"

OUTPUT_DIR=./pllava_video_outputs/test_train_7b_reconstruct

# # Naive Env
# rm -rf ${OUTPUT_DIR}
pooling_shape=(16,12,12)
accelerate launch --main_process_port 6876 --config_file scripts/accel_config_multigpu.yaml tasks/train/train_pllava_nframe_accel.py \
    tasks/train/config_pllava_nframe.py \
    output_dir ${OUTPUT_DIR} \
    train_corpus data_finance \
    save_steps 1000 \
    num_workers 4 \
    num_frames 16 \
    model.pooling_method avg \
    model.repo_id /u01/isi/luweiheng/pllava/PLLaVA/checkpoint/ermu2001/pllava-7b \
    model.use_lora False \
    model.pooling_shape $pooling_shape \
    optimizer.lr 2e-5 \
    scheduler.epochs 10 \
    scheduler.warmup_ratio 0.2 \
    scheduler.min_lr_multi 0.25 \
    scheduler.is_videochat2_custom False \
    preprocess.mm_alone False \
    preprocess.random_shuffle False \
    preprocess.add_second_msg False \`

**Comment by Annie1900** [comment]
Excuse me, I want to use a specific data set for fine-tuning, but loss remains consistent, the model doesn't seem to be training, I want to know why, here are my Settings.Thank you very much!
![8d9c6e45148fd6849c359f94083694d](https://github.com/magic-research/PLLaVA/assets/73414003/455d4656-5c70-4b35-9785-3e0456be6a93)
![image](https://github.com/magic-research/PLLaVA/assets/73414003/276984c1-4365-44d4-9f93-a0d6c3e80569)
![image](https://github.com/magic-research/PLLaVA/assets/73414003/6434bb3c-16a3-43ab-b206-4bfab8ed7586)
![image](https://github.com/magic-research/PLLaVA/assets/73414003/05c03161-43e4-44af-a435-1695d46a3543)

**Comment by Annie1900** [comment]
`from tasks.train.instruction_data import *
train_corpus = "data_finance"

train_file = "${available_corpus[${train_corpus}]}"  # for lazy evaluation
test_file = dict()
test_types = []
num_workers = 8
save_steps=1000
ckpt_steps=1000
stop_key = None
deepspeed=False
num_frames = 16
num_frames_test = 1
batch_size = 1
gradient_accumulation_steps=16
max_txt_l = 512
max_train_steps=None
pre_text = False
gradient_checkpointing=False
inputs = dict(
    image_res=336,
    video_input=dict(
        num_frames="${num_frames}",
        sample_type="rand",
        num_frames_test="${num_frames_test}",
        sample_type_test="middle",
        random_aug=False,
    ),
    max_txt_l=dict(image="${max_txt_l}", video="${max_txt_l}"),
    batch_size=dict(image="${batch_size}", video="${batch_size}"),
    batch_size_test=dict(image="${batch_size}", video="${batch_size}"),
)
model = dict(
    repo_id="checkpoint/ermu2001/pllava-7b",
    pretrained_path="checkpoint/ermu2001/pllava-7b",
    load_from_origin=False,
    origin_vision="",
    origin_llm="",
    vision_encoder=dict(
        name="vit_l14", 
    ),
    torch_dtype='bfloat16',
    freeze_projector=False,
    freeze_lm=False,
    freeze_vision_tower=True,
    lora_target_modules=["q_proj", "v_proj"], 
    use_lora=True,
    lora_r=128,
    lora_alpha=32,
    lora_dropout=0.05,
    num_frames="${num_frames}",
    pooling_method='avg',
    use_pooling=True,
    frame_shape=(24,24),
    pooling_shape=(16,8,8),
)
preprocess = dict(
    system="",
    mm_alone=True,
    random_shuffle=True,
    add_second_msg=True,
    roles=['USER:', 'ASSISTANT:'],
    end_signal=(' ', '</s>'),
    begin_signal='',
    dataset_image_placeholder='<Image></Image>',
    dataset_video_placeholder='<Video></Video>',
    image_token_index=32000,
    max_txt_l = "${max_txt_l}",
    ignore_index=-100,
    center_pad=False,
    longest_edge=762,
    shortest_edge=336,
    clip_transform=False,
    num_frames="${num_frames}",
)
optimizer = dict(
    opt="adamW",
    lr=2e-5,
    opt_betas=[0.9, 0.999],  
    weight_decay=0.02,
    max_grad_norm=-1,  
    different_lr=dict(enable=False, module_names=[], lr=1e-3),
)
scheduler = dict(
    is_videochat2_custom=False,
    sched="cosine", 
    epochs=2, 
    warmup_ratio=0.2,
    min_lr_multi=0.25)

evaluate = False
deep_fusion = False
evaluation = dict(
    eval_frame_ensemble="concat",  
    eval_x_only=False,
    k_test=128,
    eval_offload=True,  
)

fp16 = True
gradient_checkpointing = True
wandb = dict(
    enable=False,
    entity="user",
    project="videochat2",  
dist_url = "env://"
device = "cuda"
mode = "it"
output_dir = None  
resume = False  
debug = False
log_freq = 5
metric_window_size=10
seed = 42
report_to='tensorboard'
save_latest = True
auto_resume = True
pretrained_path = "" 
`

**Comment by Annie1900** [comment]
Thanks a lot!

**Comment by cathyxl** [comment]
Hi @Annie1900, please try to  change repo_id to llava-hf/llava-v1.6-vicuna-7b-hf.

**Comment by Annie1900** [comment]
> Hi @Annie1900, please try to change repo_id to llava-hf/llava-v1.6-vicuna-7b-hf.
This really helped me solve the problem, thank you very much!

**Comment by xd-liu** [comment]
Hi, how can I do continue SFT using "ermu2001/pllava-7b"

**Comment by ermu2001** [comment]
> Hi @cathyxl , how can I do continue SFT using "ermu2001/pllava-7b"?

Hi, you can now switch to branch [dev](https://github.com/magic-research/PLLaVA/tree/dev) . Although not finalized, we can now load the peft weights with MODELS/pllava-7b .

**Comment by dazhangyu123** [comment]
@xd-liu have you solved the issue?
